src/js/controller/model.js

```diff
@@ -384,8 +390,8 @@ define([
 
         };
 
-        this.persistVideoSubtitleTrack = function(trackIndex) {
-            this.setVideoSubtitleTrack(trackIndex);
+        this.persistVideoSubtitleTrack = function(trackIndex, tracks) {
+            this.setVideoSubtitleTrack(trackIndex, tracks);
```

Comment:
Isn't setting the track one of the things that triggers "subtitlesTrackChanged"? Potential loop here.